src/toolbars/ControlToolBar.cpp

```diff
@@ -615,7 +615,10 @@ void ControlToolBar::OnIdle(wxIdleEvent & event)
       return;
 
    auto &projectAudioManager = ProjectAudioManager::Get( mProject );
-   if ( projectAudioManager.Paused() )
+
+   bool isPaused = AudioIO::Get()->IsPaused();
+
+   if ( isPaused )
```

Comment:
This fix is not conservative enough.  The state changes in ProjectAudioManager no longer happen.  All the checks in that function do not happen.

I say, do not change ControlTooBar.  Rather, change ProjectAudioManager.  Make it inherit wxEvtHandler.  Then let it receive idle events from the application, similar to this other line in SpectralDataDialog.cpp:

```
   wxTheApp->Bind(wxEVT_IDLE, &SpectralDataDialogWorker::OnIdle, this);
```

Then let the event handler do all that `OnSoundActivationThreshold` did.  (Except delaying the call to Pause() using CallAfter will not be needed, as you will surely already be on the main thread and in the idle time of the event loop.)